python/tank/util/user_settings.py

```diff
@@ -36,101 +42,168 @@ def _init_singleton(self):
-        # setting was present but empty. We'll advertise that fact as None instead.
-        return self._get_value("app_store_http_proxy") or None
+        return self.get_setting(self._LOGIN, "app_store_http_proxy")
 
     @property
     def default_site(self):
         """
-        :returns: The default site.
+        Retrieves the value from the ``default_site`` setting.
         """
-        return self._get_value("default_site")
+        return self.get_setting(self._LOGIN, "default_site")
 
     @property
     def default_login(self):
         """
-        :returns: The default login.
+        Retrieves the value from the ``default_login`` setting.
         """
-        return self._get_value("default_login")
+        return self.get_setting(self._LOGIN, "default_login")
 
-    def _get_value(self, key):
+    def get_setting(self, section, name):
         """
-        Retrieves a value from the ``config.ini`` file. If the value is not set, returns the default.
-        Since all values are strings inside the file, you can optionally cast the data to another type.
+        Provides access to any setting, including ones in user defined sections.
 
-        :param key: Name of the setting within the Login section.
+        :param str section: Name of the section to retrieve the setting from. Do not include the brackets.
+        :param str name: Name of the setting under the provided section.
 
-        :returns: The appropriately type casted value if the value is found, default otherwise.
+        :returns: The setting's value if found, ``None`` if the setting is missing from the file or
+            an empty string if the setting is present but has no value associated.
+        :rtype: str
         """
-        if not self._is_setting_found(key):
+        if not self._user_config.has_section(section) or not self._user_config.has_option(section, name):
             return None
+
+        value = os.path.expanduser(
+            os.path.expandvars(
+                self._user_config.get(section, name)
+            )
+        )
+        return value.strip()
+
+    # Unfortunately here for get_boolean_setting and get_integer_setting we're replicating some of the
+    # logic from the ConfigParser class. We have to do this because ConfigParser doesn't expand environment
+    # variables which is a requirement here, so get_setting does the job of using expandvars so everything
+    # gets expanded and then the get_*_setting methods so the necessary casting.
+
+    # This is taken from RawConfigParser. Values are copied in case future Python implementation
+    # rename this. (like Python 3, not that this is going to be an issue in the foreseable future. :p)
+    _boolean_states = {'1': True, 'yes': True, 'true': True, 'on': True,
+                       '0': False, 'no': False, 'false': False, 'off': False}
+
+    def get_boolean_setting(self, section, name):
```

Comment:
Because an .ini file is a list of string value. The `get_*_settings` are used to store custom settings, which may not always be string, with some minimal validation. It seemed reasonable in this case to provide those. I could have get_setting(section, name, data_type=str) if you prefer.